File: src/main/java/com/jakobmenke/bootrestgenerator/utils/Configuration.java
package com.jakobmenke.bootrestgenerator.utils;

import com.jakobmenke.bootrestgenerator.templates.Templates;
import lombok.Data;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

@Data
public class Configuration {
    private final String srcFolder;
    private final String targetPackage;
    private final String fileName;

    public Configuration(Properties props) {
        this.fileName = props.getProperty("file.name");
        this.srcFolder = props.getProperty("target.folder");
        this.targetPackage = props.getProperty("target.package");
    }

    public static Properties readConfig(String configFileName) {
        ClassLoader classLoader = Templates.class.getClassLoader();
        InputStream in = classLoader.getResourceAsStream(configFileName);
        if (in == null) {
            return null;
        }
        Properties props = new Properties();
        try {
            props.load(in);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return props;
    }
}

File: src/main/kotlin/com/jakobmenke/bootrestgenerator/utils/Configuration.kt
package com.jakobmenke.bootrestgenerator.utils

import com.jakobmenke.bootrestgenerator.templates.Templates
import java.util.Properties

data class Configuration(
    val srcFolder: String,
    val targetPackage: String,
    val fileName: String,
    val language: String = "java"
) {
    constructor(props: Properties) : this(
        srcFolder = props.getProperty("target.folder")
            ?: defaultFolderForLanguage(props.getProperty("target.language", "java")),
        targetPackage = props.getProperty("target.package"),
        fileName = props.getProperty("file.name"),
        language = props.getProperty("target.language", "java")
    )

    companion object {
        fun defaultFolderForLanguage(language: String): String = when (language) {
            "kotlin" -> "build/generated/src/main/kotlin/"
            else -> "build/generated/src/main/java/"
        }

        fun readConfig(configFileName: String): Properties? {
            val inputStream = Templates::class.java.classLoader.getResourceAsStream(configFileName)
                ?: return null
            return Properties().apply { load(inputStream) }
        }
    }
}
